models/battleship/game_manager.go

```diff
@@ -0,0 +1,162 @@
+	GetOtherPlayerForGame(game *Game, player Player) *BattleshipPlayer
+	GetGameUuid(game *Game) string
+	GetGameDifficulty(game *Game) uint8
+	CreateHostPlayerForGame(game *Game, sessionId string) *BattleshipPlayer
+	CreateJoinPlayerForGame(game *Game, sessionId string) *BattleshipPlayer
+	SetPlayerReadyForGame(game *Game, player Player, selectedGrid Grid) error
+	AreAttackCoordinatesValid(game *Game, coordinates Coordinates) bool
+	CallRematchForGame(game *Game)
+	ResetRematchForGame(game *Game) error
+	IsRematchAlreadyCalled(game *Game) bool
+	IsGameReadyToStart(game *Game) bool
+	GetJoinPlayerSunkenShips(game *Game) uint8
+	GetHostPlayerSunkenShips(game *Game) uint8
+
+	isDifficultyValid(uint8) bool
+}
+
+type BattleshipGameManager struct {
+	games map[string]*Game
+	mu    sync.RWMutex
+}
+
+var _ GameManager = (*BattleshipGameManager)(nil)
+
+func NewBattleshipGameManager() *BattleshipGameManager {
+	return &BattleshipGameManager{
+		games: make(map[string]*Game, 10),
+	}
+}
+func (bgm *BattleshipGameManager) CreateGame(difficulty uint8) (*Game, error) {
+	if !bgm.isDifficultyValid(difficulty) {
+		return nil, cerr.ErrInvalidGameDifficulty()
+	}
+
+	gameUuid := uuid.NewString()[:6]
+	bgm.games[gameUuid] = newGame(difficulty, gameUuid)
+
+	return bgm.games[gameUuid], nil
+}
+
+func (bgm *BattleshipGameManager) GetGame(gameUuid string) (*Game, error) {
+	bgm.mu.RLock()
+	game, prs := bgm.games[gameUuid]
+	bgm.mu.RUnlock()
+	if !prs {
+		return nil, cerr.ErrGameNotExists(gameUuid)
+	}
+
+	if game == nil {
+		bgm.mu.Lock()
+		delete(bgm.games, gameUuid)
+		bgm.mu.Unlock()
+		return nil, cerr.ErrGameIsNil(gameUuid)
+	}
+
+	return game, nil
+}
+
+func (bgm *BattleshipGameManager) GetPlayer(game *Game, isHost bool) *BattleshipPlayer {
+	if isHost {
+		return game.hostPlayer
+	}
+
+	return game.joinPlayer
+}
+
+func (bgm *BattleshipGameManager) GetOtherPlayerForGame(game *Game, player Player) *BattleshipPlayer {
+	if player.IsHost() {
+		return game.joinPlayer
+	}
+	return game.hostPlayer
+}
+
+func (bgm *BattleshipGameManager) GetGameUuid(game *Game) string {
+	return game.uuid
+}
+
+func (bgm *BattleshipGameManager) GetGameDifficulty(game *Game) uint8 {
+	return game.difficulty
+}
+
+func (bgm *BattleshipGameManager) CreateHostPlayerForGame(game *Game, sessionId string) *BattleshipPlayer {
+	game.hostPlayer = newPlayer(true, true, sessionId, game.gridSize)
+	return game.hostPlayer
+}
+
+func (bgm *BattleshipGameManager) CreateJoinPlayerForGame(game *Game, sessionId string) *BattleshipPlayer {
+	game.joinPlayer = newPlayer(false, false, sessionId, game.gridSize)
+	return game.joinPlayer
+}
+
+func (bgm *BattleshipGameManager) GetHostPlayerSunkenShips(game *Game) uint8 {
+	return game.hostPlayer.sunkenShips
+}
+
+func (bgm *BattleshipGameManager) GetJoinPlayerSunkenShips(game *Game) uint8 {
+	return game.joinPlayer.sunkenShips
+}
+
+func (bgm *BattleshipGameManager) SetPlayerReadyForGame(game *Game, player Player, selectedGrid Grid) error {
```

Comment:
we're only using the `gridSize` from the game object in this function, so let's just pass the grid size instead of the whole game object.